src/Workspaces/CSharp/Portable/Utilities/CSharpUseExplicitTypeHelper.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle.TypeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.Utilities
+{
+    internal sealed class CSharpUseExplicitTypeHelper : CSharpTypeStyleHelper
+    {
+        public static CSharpUseExplicitTypeHelper Instance = new CSharpUseExplicitTypeHelper();
+
+        private CSharpUseExplicitTypeHelper()
+        {
+        }
+
+        protected override bool IsStylePreferred(
```

Comment:
Seems like the result `CSharpUseExplicitTypeHelper.IsStylePreferred` is the negation of `CSharpUseImplicitTypeHelper.IsStylePreferred`. Can we share code?

Reply:
All this code moved down.  I did not want to change it.  But i can if you think that's important.  let me know.

Reply:
Ah, I didn't realize that. In that case, I wouldn't do any rename of functionality changes as part of this PR. We can do it later. Thanks.